### PR TITLE
[Security] simplify the SwitchUserListenerTest

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        | 

While working on #22048 I noticed that the `SwitchUserListenerTest` was more complicated than necessary by mocking a lot of stuff that didn't need to be mocked.

### CODE REVIEW

**Review by GuilhemN — COMMENTED** [review]
Except my question, :+1:

File: src/Symfony/Component/Security/Http/Tests/Firewall/SwitchUserListenerTest.php [hunk 1]
```diff
@@ -253,25 +179,26 @@ public function testSwitchUser()
...
     {
-        $token = $this->getToken(array($this->getMockBuilder('Symfony\Component\Security\Core\Role\RoleInterface')->getMock()));
-        $user = $this->getMockBuilder('Symfony\Component\Security\Core\User\UserInterface')->getMock();
-        $user->expects($this->any())->method('getRoles')->will($this->returnValue(array()));
+        $token = new UsernamePasswordToken('username', '', 'key', array('ROLE_FOO'));
+        $user = new User('username', 'password', array());
 
-        $this->tokenStorage->expects($this->any())->method('getToken')->will($this->returnValue($token));
-        $this->request->expects($this->any())->method('get')->with('_switch_user')->will($this->returnValue('kuba'));
-        $this->request->query->expects($this->once())->method('remove', '_switch_user');
```
Comment (GuilhemN) [inline src/Symfony/Component/Security/Http/Tests/Firewall/SwitchUserListenerTest.php]: isn't this missing in the new test?

Comment (xabbuh) [inline src/Symfony/Component/Security/Http/Tests/Firewall/SwitchUserListenerTest.php]: We check later on that the query parameter bag is empty.

Comment (GuilhemN) [inline src/Symfony/Component/Security/Http/Tests/Firewall/SwitchUserListenerTest.php]: Ok i missed that :+1:

**Review by GuilhemN — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**